pkg/runtime/wfengine/backends/actors/activity_actor.go

```diff
@@ -131,10 +131,10 @@ func (a *activityActor) InvokeMethod(ctx context.Context, methodName string, dat
 func (a *activityActor) InvokeReminder(ctx context.Context, reminder actors.InternalActorReminder, metadata map[string][]string) error {
 	wfLogger.Debugf("Activity actor '%s': invoking reminder '%s'", a.actorID, reminder.Name)
 
-	state, _ := a.loadActivityState(ctx)
+	state, err := a.loadActivityState(ctx)
 	// TODO: On error, reply with a failure - this requires support from durabletask-go to produce TaskFailure results
-	if state == nil {
-		return errors.New("no activity state found")
+	if err != nil || len(state.EventPayload) == 0 {
+		wfLogger.Errorf("no activity state found for reminder: %s, err: %s", reminder.Name, err)
 	}
 
 	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, a.defaultTimeout)
```

Comment:
does the following logic need to be executed? 
Can't we move directly to the error switch case below or do we still support executing the workflow activity without state data?

Reply:
Based on the user seeing this err and having issues with his workflows, where their only way to recover was to delete the state table in Postgres, I believe we do want to support workflow activity without state data or at least not return here. The [prior PR](https://github.com/dapr/dapr/pull/7949) Im reverting changed the logic to return here, whereas before we weren't returning for this case.

cc @cgillum for more info here please, I see the activity state is used for the history details (event payload) saved to a statestore in case there is need to recover from a failure. I see we save state upon invoking the actor method and we purge it once the methodname is the purgeWorkflowState. I see the state gets filled with the activity request history event `ar.HistoryEvent` so it _could_ be empty after [decoding here](https://github.com/dapr/dapr/blob/master/pkg/runtime/wfengine/backends/actors/activity_actor.go#L103-L120) if the actor req data was empty to begin with. Im not familiar if this could occur and why...

For the case where it might be possible to have nil activity state, should we terminate or continue along (as we did before [this PR](https://github.com/dapr/dapr/pull/7949))?

Reply:
actually, I think `GetOrchestrationRuntimeState` in backend.go might cause the data to be empty bc I dont see it populating the data field in the req along the way and the data is used for the activity actor state. So I think it is valid and we should not return. I changed the return to a log, but will change it to debug now.

Reply:
Sounds good.